fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/SizeStatisticsCollector.java

```diff
@@ -351,30 +351,4 @@ public static SizeStatisticsCollector ofIndex(@Nonnull FDBRecordStore store, @No
     public static SizeStatisticsCollector ofSubspace(@Nonnull Subspace subspace) {
         return new SizeStatisticsCollector(subspace);
     }
-
-    /**
-     * Create a statistics collector of all keys used by index within a given {@link SubspaceProvider}'s subspace.
-     * If the implementation of {@link SubspaceProvider#getSubspace() getSubspace()} is blocking for the
-     * given <code>SubspaceProvide</code>, then this method will also be blocking.
-     *
-     * @param subspaceProvider the provider of the subspace to collect statistics on key and value sizes
-     * @return a statistics collector of the given subspace
-     */
-    @Nonnull
-    public static SizeStatisticsCollector ofSubspaceProvider(@Nonnull SubspaceProvider subspaceProvider) {
```

Comment:
Alternatively, I think this class could remember its `SubspaceProvider` instead of its `Subspace` and create the subspace when it needs it in `collectAsync`.